src/tools/cephfs/cephfs_shell.py

```diff
@@ -0,0 +1,539 @@
+#!/usr/bin/env python3
+# coding=utf-8
+
+import argparse
+import os
+import sys
+from cmd2 import Cmd, with_argparser
+import cephfs as libcephfs
+import shlex
+import shutil
+import traceback
+import colorama
+import readline
+
+cephfs =  None
+
+def mode_notation(mode):
+    permission_bits = {'0':'---', '1':'--x', '2':'-w-', '3' : '-wx', '4':'r--', '5':'r-x', '6':'rw-', '7':'rwx'}
+    mode = str(oct(mode))
+    notation = '-'
+    if mode[2] == '4':
+        notation = 'd'
+    for i in mode[-3:]:
+        notation += permission_bits[i]
+    return notation
+
+def setup_cephfs():
+    """ Mouting a cephfs """
+    global cephfs
+    cephfs = libcephfs.LibCephFS(conffile='/home/admin1/Documents/ceph/build/ceph.conf')
+    cephfs.mount()
+
+def get_file(file_name):
+    if not isinstance(file_name, bytes):
+        file_name = bytes(file_name, encoding="ascii")
+    try :
+       return cephfs.open(file_name, 'r') 
+    except : #handling file exception
+       return -1
+
+def get_chunks(file_size):
+    chunk_start = 0
+    chunk_size = 0x20000  # 131072 bytes, default max ssl buffer size
+    while chunk_start + chunk_size < file_size:
+        yield(chunk_start, chunk_size)
+        chunk_start += chunk_size
+    final_chunk_size = file_size - chunk_start
+    yield(chunk_start, final_chunk_size)
+
+def list_items(dir_name = ''):
+    if dir_name == '':
+        d = cephfs.opendir(cephfs.getcwd())
+    else :
+        d = cephfs.opendir(dir_name)
+    dent = cephfs.readdir(d)
+    items = []
+    while dent:
+        if dent.d_name not in [b".", b".."]:
+            items.append(dent)
+        dent = cephfs.readdir(d)
+    cephfs.closedir(d)
+    return items
+
+def print_details(file_name, flag):
+    fd = cephfs.fstat(get_file(file_name))
```

Comment:
This should be `cephfs.stat`? I think you're leaking a file descriptor returned from `get_file`.

Reply:
Also, `fd` is not the right name for the variable. Instead, use `buf` or `info`.